monix-eval/jvm/src/test/scala/monix/eval/TaskAsyncAutoShiftJVMSuite.scala

```diff
@@ -0,0 +1,943 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.eval
+
+import java.util.concurrent.CountDownLatch
+
+import minitest.TestSuite
+import monix.execution.exceptions.DummyException
+import monix.execution.schedulers.SchedulerService
+import monix.execution.{Cancelable, CancelableFuture, ExecutionModel, Scheduler}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+object TaskAsyncAutoShiftJVMSuite extends TestSuite[SchedulerService] {
+  private val ThreadName = "test-thread"
+
+  override def setup(): SchedulerService =
+    Scheduler.singleThread(ThreadName, executionModel = ExecutionModel.SynchronousExecution)
+
+  override def tearDown(env: SchedulerService): Unit = {
+    env.shutdown()
+    assert(env.awaitTermination(10.seconds), "env.awaitTermination")
+  }
+
+  def createFuture[A](ec: ExecutionContext)(r: => A): (Future[A], CountDownLatch) = {
+    val started = new CountDownLatch(1)
+    val latch = new CountDownLatch(1)
+    val f = Future {
+      started.countDown()
+      latch.await()
+      r
+    }(ec)
+
+    started.await()
+    (f, latch)
+  }
+
+  def createCancelableFuture[A](ec: ExecutionContext)(r: => A): (CancelableFuture[A], CountDownLatch) = {
+    val started = new CountDownLatch(1)
+    val latch = new CountDownLatch(1)
+
+    val f = Future {
+      started.countDown()
+      latch.await()
+      r
+    }(ec)
+
+    started.await()
+    (CancelableFuture(f, Cancelable.empty), latch)
+  }
+
+  def repeatTest(times: Int)(f: => Future[Unit])(implicit ec: ExecutionContext): Future[Unit] = {
+    if (times > 0) f.flatMap(_ => repeatTest(times - 1)(f))
+    else Future.successful(())
```

Comment:
nitpick: `Future.unit`

Reply:
Isn't that 2.12+

Reply:
if that's a case, disregard my comment :D

Reply:
Yeah, `Future.unit` is not available in 2.11 ☹️